app/models/concerns/operations/stock_operations.rb

```diff
@@ -5,6 +5,43 @@ module StockOperations
 
     module_function
 
+
+    ##
+    # Adds a package to the inventory
+    #
+    #
+    # @raise [Goodcity::AlreadyInventorizedError] thrown if trying to inventorize twice
+    #
+    # @param [Package|String] the package to inventorize or its id
+    # @param [Location|String] the location to place the package in
+    #
+    def inventorize(package, location)
+      last = PackagesInventory.order('id DESC').where(package: package).first
```

Comment:
@patrixr, Just a suggestion: 

`PackagesInventory.order('id DESC').where(package: package)` In this, it will order all table records first then apply where query. Eventually, we will have large number of records inside this table. So I feel first applying where query and then doing order will be better.

`PackagesInventory..where(package: package).order('id DESC').first`

Let me know your thoughts. :)

Reply:
I don't actually think it makes a difference.
ActiveRecord compiles it into SQL right before running it so it wouldn't change anything. You can check that by doing `to_sql` on the query. It will return the same one

Although it's true that it could be slow **IF** `first` is an array method rather than a relation method (actually likely now that I think about it), in which case I think `limit(1)` is probably the optimisation we need

Reply:
When chaining AREL queries, it generally doesn't matter what order you write the clauses in as it will create an abstract syntax tree and combine clauses where necessary. So it's fine for `order` to come before `where` if desired. Just did a test with `first` and confirmed that just does a `limit(1)` and `order('id')` and then returns an instance of the the underlying model class. See the SQL below, which has `ORDER BY` and `LIMIT`

>> Order.first
  Order Load (1.4ms)  SELECT  "orders".* FROM "orders"  ORDER BY "orders"."id" ASC LIMIT 1
 => #<Order id: 1, status: "closed"...>